pkg/skaffold/build/cache/hash_test.go

```diff
@@ -195,3 +225,109 @@ func TestCacheHasher(t *testing.T) {
 		})
 	}
 }
+
+func TestRetrieveBuildArgs(t *testing.T) {
+	tests := []struct {
+		description  string
+		artifactType latest.ArtifactType
+		expected     map[string]*string
+	}{
+		{
+			description: "docker artifact with build args",
+			artifactType: latest.ArtifactType{
+				DockerArtifact: &latest.DockerArtifact{
+					BuildArgs: map[string]*string{},
+				},
+			},
+			expected: map[string]*string{},
+		}, {
+			description: "docker artifact without build args",
+			artifactType: latest.ArtifactType{
+				DockerArtifact: &latest.DockerArtifact{},
+			},
+		}, {
+			description: "kaniko artifact with build args",
+			artifactType: latest.ArtifactType{
+				KanikoArtifact: &latest.KanikoArtifact{
+					BuildArgs: map[string]*string{},
+				},
+			},
+			expected: map[string]*string{},
```

Comment:
Can we add some build args and test the expected values?

Reply:
I do that it `TestGetHashForArtifact`, this is just to make sure we get the build args if they are defined.